site/content/en/contributions/design/backend.md

```diff
@@ -0,0 +1,159 @@
+* In some cases, Envoy Gateway may reject references to a `Backend` resource. For example, a backend with a UDS address 
+  will be rejected on xRoute references. 
+* All attributes of the Envoy Gateway extended `BackendRef` resource MUST be implemented for the `Backend` resource.  
+* A `Backend` resource referenced by `BackendObjectReference` will be translated to Envoy Gateway's IR DestinationSetting.
+  As such, all `BackendAdresses` are treated as equivalent endpoints with identical weights, TLS settings, etc.  
+* Gateway-API and Envoy Gateway policies that attach to Services ([BackendTLSPolicy][], [BackendLBPolicy][]) 
+  MUST support attachment to the `Backend` resource in Envoy Gateway. 
+* Policy attachment to a named section of the `Backend` resource is not supported at this time. Currently, 
+  `BackendObjectReference` can only select ports, and not generic section names. Hence, a named section of `Backend` 
+  cannot be referenced by routes, and so attachment of policies to named sections will create translation ambiguity. 
+  Users that wish to attach policies to some of the `BackendAddresses` in a `Backend` resource can use multiple `Backend` 
+  resources and pluralized `BackendRefs` instead. 
+* The `Backend` API SHOULD support other Gateway-API backend features, such as [Backend Protocol Selection][]. 
+  Translation of explicit upstream application protocol setting MUST be consistent with the existing implementation for
+  `Service` resources. 
+* This API resource MUST be part of same namespace as the targetRef resource. The `Backend` API MUST be subject to 
+  the same cross-namespace reference restriction as referenced `Service` resources.    
+* The `Backend` resource translation MUST NOT modify Infrastructure. Any change to infrastructure that is required to 
+  achieve connectivity to a backend (mounting a socket, adding a sidecar container, modifying a network policy, ...) 
+  MUST be implemented with an appropriate infrastructure patch in the [EnvoyProxy][] API. 
+* To limit the overall maintenance effort related to supporting of non-Service backends, the `Backend` API SHOULD 
+  support multiple generic address types (UDS, FQDN, IPv4, IPv6), and MUST NOT support vendor-specific backend types.
+* Both `Backend` and `Service` resources may appear in the same `BackendRefs` list.
+* The Optional `Port` field SHOULD NOT be evaluated when referencing a `Backend`.  
+* Referenced `Backend` resources MUST be translated to envoy endpoints, similar to the current `Service` translation.
+* Certain combinations of `Backend` and `Service` are incompatible. For example, a Unix Domain Socket and a FQDN service
+  require different cluster service discovery types (Static/EDS and Strict-DNS accordingly).
+* If a Backend that is referenced by a route cannot be translated, the `Route` resource will have an `Accepted=False` 
+  condition with a `UnsupportedValue` reason. 
+  
```

Comment:
I'd vote to make this opt in at the `EnvoyGateway` level by an admin to avoid
https://github.com/advisories/GHSA-vw47-mr44-3jf9